### DISCUSSION TITLE
same file with different compile command in one compile_commands.json

### DISCUSSION THREAD

**Discussion by tarik-jin** [clangd/clangd#1288] [Ideas]
seems clangd uses first entry it encounters, what about popup an window and let user to choose which compile command to interpret for the current open file.

**Comment by rhoot** [comment]
Personally I'd like it to parse all of them, even if it has to do it multiple times and uses more resources. It would be cool if I could e.g. have two build configurations&mdash;one with `NDEBUG` defined and one without&mdash;and have this code trigger an unused variable warning regardless of the "active" config (they'd all be active at that point).

```cpp
#include <cassert>
int someFunction();
void test()
{
    int i = someFunction(); // warn about unused variable `i`
    assert(i == 0);
}
```

I generally build with fatal warnings, so that would let me catch those warnings as I'm working on the code rather than having to compile all configs to find all of them. It would also allow the editor to show errors in all potentially active pre-processor blocks simultaneously, and that inactive pre-processor blocks are _truly_ inactive (not used in _any_ config).

**Reply by HighCommander4** [reply]
Sounds like #681

**Reply by rhoot** [reply]
So it does. Thanks for the link, I'll keep an eye on that issue. :+1:
